### PR TITLE
DAOSGCP-163 IO500 example refactoring

### PR DESCRIPTION
**Pull request by mark-olson**
Refactored terraform/examples/io500

Changes starting with '[ML]' were either contributed directly or suggested by Margaret Lawson.

- Wrapper scripts moved to `bin/` directory
- Client scripts moved to `client_scripts/` directory
- Scripts in `images/` directory updated to wrap new `images/build.sh` script
- Do not use terraform/examples/daos_cluster for io500 example. terraform/examples/io500 has its own terraform configuration now.
- Removed the use of TF_VARS_* environment variables. Now uses a terraform.tfvars file that is generated based on the config/*.sh file that is used by the start.sh script.
- [ML] SERVER_OS_FAMILY and CLIENT_OS_FAMILY adjusted to include the word "DAOS"
- [ML] Do not use preemptible instances for IO500

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

Signed-off-by: Mark Olson <115657904+mark-olson@users.noreply.github.com>

### CODE REVIEW

File: terraform/examples/io500/README.md [hunk 1]
```diff
@@ -1,51 +1,145 @@
...
+| <a name="input_network_name"></a> [network\_name](#input\_network\_name) | Name of the GCP network | `string` | `"default"` | no |
+| <a name="input_project_id"></a> [project\_id](#input\_project\_id) | The GCP project | `string` | n/a | yes |
+| <a name="input_region"></a> [region](#input\_region) | The GCP region | `string` | n/a | yes |
+| <a name="input_server_daos_crt_timeout"></a> [server\_daos\_crt\_timeout](#input\_server\_daos\_crt\_timeout) | crt\_timeout | `number` | `300` | no |
+| <a name="input_server_daos_disk_count"></a> [server\_daos\_disk\_count](#input\_server\_daos\_disk\_count) | Number of local ssd's to use | `number` | `16` | no |
+| <a name="input_server_daos_scm_size"></a> [server\_daos\_scm\_size](#input\_server\_daos\_scm\_size) | scm\_size | `number` | `200` | no |
+| <a name="input_server_gvnic"></a> [server\_gvnic](#input\_server\_gvnic) | Use Google Virtual NIC (gVNIC) network interface | `bool` | `false` | no |
+| <a name="input_server_instance_base_name"></a> [server\_instance\_base\_name](#input\_server\_instance\_base\_name) | MIG instance base names to use | `string` | `"daos-server"` | no |
```
Comment (mlawsonca) [inline terraform/examples/io500/README.md:127]: Should this say MIG?

File: terraform/examples/io500/variables.tf [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+}
+
+variable "client_machine_type" {
+  description = "GCP machine type. ie. c2-standard-16"
+  default     = "c2-standard-16"
+  type        = string
+}
+
+variable "client_instance_base_name" {
+  description = "MIG instance base names to use"
```
Comment (mlawsonca) [inline terraform/examples/io500/variables.tf:213]: Again, should MIGs be mentioned?

**Review by mlawsonca — COMMENTED** [review]
Like you mentioned it is a fairly large PR but on first glance it looks good. My only comment is about the references to MIGs found throughout the code.

**Review by ravalsam — DISMISSED** [review]
Verified all the test cases described, and all worked fine.

File: terraform/examples/io500/README.md (hunk 1 above)
Comment (mark-olson) [reply terraform/examples/io500/README.md]: Removed any text that refers to "MIG"

File: terraform/examples/io500/variables.tf (hunk 2 above)
Comment (mark-olson) [reply terraform/examples/io500/variables.tf]: Removed any text that refers to "MIG"

**Review by ravalsam — APPROVED** [review]

**Pull request merged**